### ISSUE TITLE
Failed to ensure the required bucket exist

### ISSUE THREAD

**Issue by kcatstack** [spinnaker/spinnaker#4312]
### Issue Summary:

I'm trying to set up automated canary analysis with prometheus metric store and gcs storage service. I get `Failed to ensure the required bucket exist` when I try to enable gcs in google account

### Cloud Provider(s):

Kubernetes V2

### Environment:

GKE

### Feature Area (if this issue is UI/UX related, please tag `@spinnaker/ui-ux-team`):

Halyard

### Steps to Reproduce:

Follow https://www.spinnaker.io/setup/canary/
1. `hal config canary enable`
2. `hal config canary google enable`
3. 
```
hal config canary google account add my-google-account \
  --project $PROJECT_ID \
  --json-path $JSON_PATH \
  --bucket $MY_SPINNAKER_BUCKET
```
4. `hal config canary google edit --gcs-enabled true`
Here I get:
```
Problems in default.canary:
! ERROR Failed to ensure the required bucket
  "my-spinnaker-bucket" exists: null
```
I'm sure the gcs bucket I specified exist

**Comment by kcatstack** [comment]
Ok. Updated from me here.
I faced the issue when updating halyard from 1.17 to 1.19
To get around this issue. I have downgraded the halyard version to 1.17
Anyone know which might cause it?

**Comment by milesmatthias** [comment]
I'm seeing this in Halyard 1.18 / Spinnaker 1.13.1.

**Comment by duftler** [comment]
Was able to reproduce with Halyard 1.19.2 and 1.18.0.
As reported, I do not see the issue with 1.17.0.
Looking into it now.

**Comment by duftler** [comment]
fyi @plumpy

**Comment by rafmonteiro** [comment]
One thing that I noticed is that if I don't set the region and the endpoint I'll have the same issue, but that's for AWS S3 not sure if that helps with gcs.

**Comment by duftler** [comment]
Verified that the fix is in 1.20.2 and that it behaves as expected.
